File: C++/167_2Sum2_2Pointers.cpp
#include "package_wwq.cpp"
/*
Computational complexity: O(n)
Space: O(1)
*/
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size();
        int l=0, r=n-1;
        vector<int> res;
        
        while(l<r){
            int s = numbers[l]+numbers[r];
            if (s>target){
                r--;
            }
            else if(s< target){
                l++;
            }
            else{
                res.push_back(l+1);
                res.push_back(r+1);
                return res;
            }
        }
        return res;
    }
};

int main(){
    vector<int> numbers{2, 7, 11, 15};
    int target = 9;
    
    vector<int> res = Solution().twoSum(numbers, target);
    for (int i=0; i< res.size(); i++)
        cout << res[i] << ',';
    cout << endl;
}

File: Python/167_2Sum2_2Pointers.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        l, r = 0, len(numbers)-1
    	res =[]
        while(l<r):
        	s = numbers[l]+numbers[r]
        	if s < target:
        		l+=1
        	elif s > target:
        		r-=1
        	else:
        		res.append(l+1)
        		res.append(r+1)
        		return res
        return res

